### ISSUE TITLE
Ugly formatting html in .vue file

### ISSUE THREAD

**Issue by oswaldofreitas** [prettier/prettier#5844]
I just updated prettier in my vue project and now in a SFC .vue file it's formatting this code:
```
<template>
  <router-link
    class="margin-right-medium"
    to="/admin/ads/list">Lista</router-link>
</template>
```
into this:
```
<template>
  <router-link class="margin-right-medium" to="/admin/ads/list"
    >Lista</router-link
  >
</template>
```
this is very ugly while I expected something like this:
```
<template>
  <router-link
    class="margin-right-medium"
    to="/admin/ads/list">
    Lista
  </router-link>
</template>
```

**Comment by j-f1** [comment]
We’re unable to do that because adding whitespace there would change how the HTML renders.

**Comment by duailibe** [comment]
Oswaldo, you might be interested in the `--html-whitespace-sensitivity` option:

**Prettier 1.16.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeeBbADgGwIbwB8AOlAARmoBOEArvFQLQ4CWUA1qRRWPgM58AvMRAY8VAOZtGVFhIAWMRhjgATFrQwiu3GBGEgA9HlUY2x1X0Os+MEYQAyLW3lSGa9OE1YcSUN5i4BHB+IAA0IBBYMCzQfMig4jQA7gAK4gjxKHgAbhAsquEgAEZUeGDscDAAyljlbBLIMFS0cBGKGDgA6vIs8Hx1YHDVmX0sOX0AnsggclAQVHBFbHxeMKllEmLIAGZ4OKsRAFZ8AB4AQmUVVdV4Kk5QS0h7B20gJ6fVDThwAIq0EHgu32hxAdSoqyoMxyrSKWFksC6BRg8mQAA4AAwReEQVZdMpYGbwuCQnJLCKLACOtBYiw2eC2eGBrwiqzMTRabz43z+AKBzxBbxgeGKSNUKOQACYIs08CwfBIAMIQDDbFDzR5FWirAAqIqyL1WAF8jUA)
```sh
--html-whitespace-sensitivity ignore
--parser vue
```

**Input:**
```vue
<template>
  <router-link
    class="margin-right-medium"
    to="/admin/ads/list">Lista</router-link>
</template>

```

**Output:**
```vue
<template>
  <router-link class="margin-right-medium" to="/admin/ads/list">
    Lista
  </router-link>
</template>

```

**Comment by oswaldofreitas** [comment]
@duailibe thanks for your suggestion, but when I changed my .eslintrc.js to this:
```
    ...
    'prettier/prettier': [
      'error',
      {
        semi: false,
        singleQuote: true,
        trailingComma: 'all',
        html-whitespace-sensitivity: 'ignore',
        parser: 'vue'
      },
    ],
```
it messed up my .js files and didn't solve the closing bracket in a new line. This new line for a closing bracket hurts my eyes when reading code, this is like a lot of new symbols into my code.

**Comment by lydell** [comment]
@oswaldofreitas `parser: "vue"` won't work with JS files :)

**Comment by oswaldofreitas** [comment]
@lydell anyway it didn't fix the ">" in a new line. I'm not sure how to specify this `parser: "vue"` option only for vue files in a project created by vue cli

**Comment by j-f1** [comment]
> I'm not sure how to specify this `parser: "vue"` option only for vue files in a project created by vue cli

You shouldn’t need to. Prettier will auto-detect which parser to use based on the file name.

**Comment by oswaldofreitas** [comment]
thanks guys, so there is any workaround I can use to keep my vue packages updated but disable new prettier rules until this format is fixed?

**Comment by lydell** [comment]
Prettier isn't composed of rules in that way, it's more of a all-or-nothing type of thing. Note that if you manually add some spaces/newlines in certain places Prettier will preserve (some) of them (because it can't know what is safe to do). For example:

**Prettier 1.16.4**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeeBbADgGwIbwB8AOlAARmoBOEArvFQLQ4CWUA1qRRWPgM58AvMRAY8VAOZtGVFhIAWMRhjgATFrQwiu3GBGEgA9HlUY2x1X0Os+MESXLcAMi1t4dqQzXpwmrDg6emLgEcA4gADQgEFgwLNB8yKDiNADuAAriCIkoeABuECyqkSAARlR4YOxwMADKWJVsEsgwVLRwUYoYOADq8izwfA1gcLXZAyx5AwCeyOACJWx8vjDpFRJiyABmeDjLUQBWfAAeAEIVVTW1eCouUHDbu-sgR8e1TThwAIq0EPCPew6IAaVGWVDmeXaJSwslgPSKMHkyAAHAAGKIwiDLHoVLBzGFwMF5B5RKhwACOtBYZLWeA2eABz2WZhabSBfA+31+-yQO0BURgeFK8NUiOQACYBRUWP4JABhCAYTYoKDQEkgWjLAAqQpyfOWAF8DUA)
```sh
--parser vue
```

**Input:**
```vue
<template>
  <router-link
    class="margin-right-medium"
    to="/admin/ads/list">
    Lista
  </router-link>
</template>

```

**Output:**
```vue
<template>
  <router-link class="margin-right-medium" to="/admin/ads/list">
    Lista
  </router-link>
</template>

```

But I do agree that splitting the end tag (`</router-link`) across several lines is a little bit unusual. It's no bug though – Prettier has always done so for HTML. Maybe @ikatyang knows where this comes from?

**Comment by oswaldofreitas** [comment]
here is my .eslintrc.js
```
module.exports = {
  root: true,
  env: {
    node: true,
  },
  extends: ['plugin:vue/recommended', '@vue/prettier'],
  plugins: ['sort-imports-es6-autofix'],
  rules: {
    curly: 'error',
    'no-console': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'no-mixed-operators': 'error',
    'no-param-reassign': 'error',
    'no-restricted-imports': [
      'error',
      {
        patterns: ['./*', '../*'],
      },
    ],
    'prettier/prettier': [
      'error',
      {
        semi: false,
        singleQuote: true,
        trailingComma: 'all',
      },
    ],
    'sort-imports-es6-autofix/sort-imports-es6': 'error',
    'vue/attribute-hyphenation': ['error', 'always'],
    'vue/attributes-order': [
      2,
      {
        order: [
          'GLOBAL', // id
          'DEFINITION', // is
          'UNIQUE', // ref, key, slot
          'CONDITIONALS', // v-if, v-else-if, v-else, v-show, v-cloak
          'LIST_RENDERING', // v-for
          'OTHER_ATTR', // all unspecified bound & unbound attributes
          'BINDING', // v-model
          'RENDER_MODIFIERS', // v-pre, v-once
          'CONTENT', // v-html, v-text
          'EVENTS', // v-on
        ],
      },
    ],
    'vue/component-name-in-template-casing': ['error', 'kebab-case'],
    'vue/html-end-tags': 'error',
    'vue/html-self-closing': [
      'error',
      {
        html: {
          void: 'always',
          normal: 'always',
          component: 'always',
        },
        svg: 'always',
        math: 'always',
      },
    ],
    'vue/name-property-casing': ['error', 'kebab-case'],
    'vue/no-multi-spaces': 'error',
    'vue/order-in-components': [
      'error',
      {
        order: [
          'el',
          'name',
          'parent',
          'functional',
          ['delimiters', 'comments'],
          ['components', 'directives', 'filters'],
          'extends',
          'mixins',
          'inheritAttrs',
          'model',
          ['props', 'propsData'],
          'data',
          'computed',
          'watch',
          'LIFECYCLE_HOOKS',
          'methods',
          ['template', 'render'],
          'renderError',
        ],
      },
    ],
    'vue/require-prop-types': ['error'],
    'vue/require-valid-default-prop': ['error'],
  },
  parserOptions: {
    parser: 'babel-eslint',
  },
}
```

**Comment by axit-joost** [comment]
Hi @oswaldofreitas,

I got the same issue today, with the `</closing-tags>` that the closing `>` was on a new line.

I ended up creating a `.prettierrc` with that `htmlWhitespaceSensitivity` set to ignore. That seemed to be picked up by my eslint + prettier setup and formatted everything correctly.

So at least for my, @duailibe 's suggestion he made [here](https://github.com/prettier/prettier/issues/5844#issuecomment-462309084) worked for me. I didn't even bother with adding  `parser: vue` anywhere.
